migration/migrator/migrations/system/20220417174632_patch_authentication_method.py

```diff
@@ -0,0 +1,29 @@
+"""Migration for the Submitty system."""
+from collections import OrderedDict
+import json
+
+def up(config):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    """
+    authentication_file = config.config_path / 'authentication.json'
+    with authentication_file.open('r+') as auth_file:
+        auth_info = json.load(auth_file, object_pairs_hook=OrderedDict)
+        if auth_info['ldap_options'] == []:
```

Comment:
Better to do [`isinstance](https://docs.python.org/3/library/functions.html#isinstance) check for type as if `auth['ldap_options']` was defined to a list with elements, then this equality would fail. Additionally, since we want the value to be a `dict` and everything else is wrong, I'd make this more "broad" and just do:

```suggestion
        if not isinstance(auth_info['ldap_options'], dict):
```